samples/extensions/sparse_image/sparse_image.h

```diff
@@ -0,0 +1,375 @@
+		std::set<TextureBlock> texture_block_update_set;
+
+		// Set containing information which pages from the page_table should be updated (either loaded from CPU memory or blitted)
+		std::set<size_t> update_set;
+
+		// Vector of available memory pages (whole memory page pool is statically allocated at the beginning)
+		std::vector<size_t> available_memory_page_indices;
+
+		// Sparse-image-related format and memory properties
+		VkSparseImageFormatProperties   format_properties;
+		VkSparseImageMemoryRequirements memory_sparse_requirements;
+		VkMemoryRequirements            mem_requirements;
+
+		std::vector<VkSparseImageMemoryBind> sparse_image_memory_bind;
+	};
+
+	struct CalculateMipLevelData
+	{
+		std::vector<std::vector<Point>>    mesh;
+		std::vector<std::vector<MipBlock>> mip_table;
+
+		uint32_t vertical_num_blocks;
+		uint32_t horizontal_num_blocks;
+
+		uint8_t mip_levels;
+
+		std::vector<float> ax_vertical;
+		std::vector<float> ax_horizontal;
+
+		glm::mat4 mvp_transform;
+
+		VkExtent2D texture_base_dim;
+		VkExtent2D screen_base_dim;
+
+		CalculateMipLevelData(const glm::mat4 &mvp_transform, const VkExtent2D &texture_base_dim, const VkExtent2D &screen_base_dim, uint32_t vertical_num_blocks, uint32_t horizontal_num_blocks, uint8_t mip_levels);
+		CalculateMipLevelData();
+
+		void calculate_mesh_coordinates();
+		void calculate_mip_levels();
+	};
+
+	// UI related
+	bool color_highlight         = true;
+	bool color_highlight_changed = false;
+	bool memory_defragmentation  = true;
+	bool frame_counter_feature   = true;
+
+	size_t blocks_to_update_per_cycle = 10U;
+
+	size_t num_vertical_blocks   = 50U;
+	size_t num_horizontal_blocks = 50U;
+
+	size_t num_vertical_blocks_upd   = 50U;
+	size_t num_horizontal_blocks_upd = 50U;
+
+	bool update_required = false;
+
+	uint8_t frame_counter_per_transfer = 0U;
+
+	const uint8_t FRAME_COUNTER_CAP        = 10U;
+	const uint8_t MEMORY_FRAGMENTATION_CAP = 20U;
+	const uint8_t PAGES_PER_ALLOC          = 50U;
+	const double  FOV_DEGREES              = 60.0;
+
+	Stages next_stage = Stages::Idle;
+
+	const VkFormat    image_format = VK_FORMAT_R8G8B8A8_SRGB;
+	VkImageUsageFlags image_usage  = VK_IMAGE_USAGE_TRANSFER_DST_BIT | VK_IMAGE_USAGE_TRANSFER_SRC_BIT | VK_IMAGE_USAGE_SAMPLED_BIT;
+
+	VirtualTexture virtual_texture;
+
+	CalculateMipLevelData mesh_data;
+
+	VkQueue sparse_queue;
+
+	std::unique_ptr<vkb::core::Buffer> vertex_buffer;
+
+	std::unique_ptr<vkb::core::Buffer> index_buffer;
+	size_t                             index_count;
+
+	std::unique_ptr<vkb::core::Buffer> mvp_buffer;
+	std::unique_ptr<vkb::core::Buffer> frag_settings_data_buffer;
+
+	glm::mat4 current_mvp_transform;
+
+	VkPipeline       sample_pipeline{};
+	VkPipelineLayout sample_pipeline_layout{};
+
+	VkDescriptorSetLayout descriptor_set_layout;
+	VkDescriptorSet       descriptor_set;
+	VkSampler             texture_sampler;
+
+	VkSemaphore bound_semaphore;
+	VkSemaphore submit_semaphore;
```

Comment:
Some Vk-member variables are default-initialized, some are not?

Reply:
Removed default initialization to keep it consistent.